utilities/transactions/write_prepared_txn.cc

```diff
@@ -100,7 +108,7 @@ Status WritePreparedTxn::CommitBatchInternal(WriteBatch* batch) {
                                no_log_ref, !disable_memtable, &seq_used);
   assert(seq_used != kMaxSequenceNumber);
   uint64_t& prepare_seq = seq_used;
-  uint64_t& commit_seq = seq_used;
+  uint64_t commit_seq = GetACommitSeqNumber(prepare_seq);
```

Comment:
Does this write batch come with prepare and commit markers or does it appear as a non-transactional write? If it is the former, will increment of last_to_be_written_seq leave a hold in WAL and break recovery?

Reply:
It is the latter. WriteBatch and CommitWithoutPrepare do not have a prepare or commit marker in the WAL.

Reply:
I mean if it is the latter, it seems to me it will leave a hole in WAL, will it?

Reply:
True. And the hole might be problematic: https://github.com/facebook/rocksdb/pull/3046#discussion_r146079740  I am looking into it.